### PR TITLE
Refactor: use go-fx, pangu & software architecture

### PR DESCRIPTION
**Pull request by AH-dark**
Use a more standardized structural framework and a more superior technology stack.

- Use [go-fx](https://github.com/uber-go/fx) for dependency injection to avoid confusion caused by too many global variables.
- Use [samber/lo](https://github.com/samber/lo) for better lambda-like operations.

The above modifications enable the application to continue to expand, enabling contributors in the open source community to understand the meaning of fadian better and faster, and also make more people get into a state of fadian.

### CODE REVIEW

**Review by purofle — APPROVED** [review]
looks 没问题

**Comment by AH-dark** [comment]
> looks 没问题

So should we consider merging into the main branch? 🤨

**Comment by AH-dark** [comment]
> looks 没问题

另外，为什么 JVM 大佬会来审核 Go 的代码？🤔

File: service/fabing/init.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+	"go.uber.org/zap"
+
+	"github.com/afadian/fadian-go/internal/util"
+)
+
+type BaseText string
+
+func Initialize() (BaseText, error) {
+	f, err := os.ReadFile(util.AbsolutePath("data/sections/fabing.txt"))
```
Comment (2096779623) [inline service/fabing/init.go:14]: `go build`之后移动到任意位置之后会报错，不能作为单文件运行，能不能把文字打包成变量/运行的时候输出到文件？这样会方便一点

**Comment by chitang233** [comment]
> > looks 没问题
> 
> 另外，为什么 JVM 大佬会来审核 Go 的代码？🤔

这就是大佬

**Review by 2096779623 — CHANGES_REQUESTED** [review]

**Comment by AH-dark** [comment]
> > > looks 没问题
> > 
> > 
> > 另外，为什么 JVM 大佬会来审核 Go 的代码？🤔
> 
> 这就是大佬

🧎‍♂️🧎‍♂️🧎‍♂️

File: service/fabing/init.go (hunk 1 above)
Comment (AH-dark) [reply service/fabing/init.go:14]: Oh, it was my mistake, please wait a moment.

Comment (AH-dark) [reply service/fabing/init.go]: I resolved this problem in commit https://github.com/afadian/fadian-go/pull/1/commits/a4bdbb33072e7c49c12a822c043532b93c244b87

**Comment by AH-dark** [comment]
Also, I think embedding the file into the program might not be a good solution, maybe we can put the file in the root directory and allow users to modify the file without compiling it themselves?

File: service/fabing/init.go (hunk 1 above)
Comment (AH-dark) [reply service/fabing/init.go]: 我盲猜日落果压根没跑代码就 Approved @purofle

**Comment by 2096779623** [comment]
> Also, I think embedding the file into the program might not be a good solution, maybe we can put the file in the root directory and allow users to modify the file without compiling it themselves? @2096779623

This is also a good choice.👍

**Comment by AH-dark** [comment]
Maybe I can implement this idea in another pull request.

**Comment by 2096779623** [comment]
> Maybe I can implement this idea in another pull request.

Do what you want to do.

**Comment by 2096779623** [comment]
> Maybe I can implement this idea in another pull request.

edit: Preferably after this PR is merged.

**Comment by 2096779623** [comment]
有个变量逃逸：
```
~/fadian-go $ go build -o fadian -gcflags=-m
# github.com/afadian/fadian-go
./main.go:19:16: inlining call to flag.StringVar
./main.go:19:16: inlining call to flag.newStringValue
./main.go:20:14: inlining call to flag.BoolVar
./main.go:20:14: inlining call to flag.newBoolValue
./main.go:21:14: inlining call to flag.BoolVar
./main.go:21:14: inlining call to flag.newBoolValue
./main.go:22:13: inlining call to flag.IntVar
./main.go:22:13: inlining call to flag.newIntValue
./main.go:23:14: inlining call to flag.BoolVar
./main.go:23:14: inlining call to flag.newBoolValue
./main.go:24:12: inlining call to flag.Parse
./main.go:29:6: can inline main
./main.go:14:26: inlining call to context.Background
./main.go:18:15: &data.BootstrapParams{} escapes to heap
~/fadian-go $
```

**Comment by AH-dark** [comment]
> 有个变量逃逸：
> 
> ```
> ~/fadian-go $ go build -o fadian -gcflags=-m
> # github.com/afadian/fadian-go
> ./main.go:19:16: inlining call to flag.StringVar
> ./main.go:19:16: inlining call to flag.newStringValue
> ./main.go:20:14: inlining call to flag.BoolVar
> ./main.go:20:14: inlining call to flag.newBoolValue
> ./main.go:21:14: inlining call to flag.BoolVar
> ./main.go:21:14: inlining call to flag.newBoolValue
> ./main.go:22:13: inlining call to flag.IntVar
> ./main.go:22:13: inlining call to flag.newIntValue
> ./main.go:23:14: inlining call to flag.BoolVar
> ./main.go:23:14: inlining call to flag.newBoolValue
> ./main.go:24:12: inlining call to flag.Parse
> ./main.go:29:6: can inline main
> ./main.go:14:26: inlining call to context.Background
> ./main.go:18:15: &data.BootstrapParams{} escapes to heap
> ~/fadian-go $
> ```

fixed.

**Comment by 2096779623** [comment]
往上看看还有readme.md要改的

**Comment by AH-dark** [comment]
> 往上看看还有readme.md要改的

Please tell me in detail. 🤨

**Comment by 2096779623** [comment]
>  Please tell me in detail. 🤨

```diff
--- README.md.orig      2023-01-27 20:05:18.942275071 +0800
+++ README.md   2023-01-27 20:06:40.482275040 +0800
@@ -18,6 +18,7 @@
 ## Build with source code

 1. Clone this repository: `git clone https://github.com/afadian/fadian-go.git`
-2. Run `go mod download` to download dependencies
-3. Run `go build -o fadian` to build the project
-4. Run `./fadian` to run the project
+2. Run `cd fadian-go` to enter directory
+3. Run `go mod download` to download dependencies
+4. Run `go build -o fadian-go` to build the project
+5. Run `./fadian-go` to run the project
```

**Review by 2096779623 — APPROVED** [review]

**Pull request merged**